core/signal/signame_spec.rb

```diff
@@ -0,0 +1,25 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+describe "Signal.signame" do
+  it "takes a signal name with a well known signal number" do
+    Signal.signame(0).should == "EXIT"
+    Signal.signame(1).should == "HUP"
+    Signal.signame(2).should == "INT"
+  end
```

Comment:
Yes, it's a trade-off. If the numbers are portable on POSIX platforms, it's OK to use hardcoded numbers.